code/game/objects/items/reagent_containers/hypospray.dm

```diff
@@ -296,27 +252,45 @@
 		user.balloon_alert(user, "You tag [src] as \"[str]\".")
 		description_overlay = str
 		update_icon()
+		return TRUE
 
-	else if(href_list["set_transfer"])
+	if(action == "ToggleMode")
+		if(inject_mode)
+			to_chat(usr, span_notice("[src] has been set to draw mode. It will now drain reagents."))
+
+		else
+			to_chat(usr, span_notice("[src] has been set to inject mode. It will now inject reagents."))
+		inject_mode = !inject_mode
+		update_icon()
+		return TRUE
+
+	if(action == "SetTransferAmount")
 		var/N = tgui_input_list(usr, "Amount per transfer from this:", "[src]", possible_transfer_amounts)
 		if(!N)
 			return
 
 		amount_per_transfer_from_this = N
+		return TRUE
 
-	else if(href_list["flush"])
+	if(action == "EmptyHypospray")
 		empty(usr)
+		return TRUE
 
-	updateUsrDialog()
-
-
-/obj/item/reagent_containers/hypospray/advanced/Topic(href, href_list)
+/obj/item/reagent_containers/hypospray/advanced/ui_act(action, list/params, datum/tgui/ui, datum/ui_state/state)
 	. = ..()
 	if(.)
 		return
 
-	if(href_list["displayreagents"])
+	if(action == "DisplayReagentContent")
 		to_chat(usr, display_reagents())
+		return TRUE
+
+/obj/item/reagent_containers/hypospray/advanced/ui_interact(mob/user, datum/tgui/ui)
```

Comment:
Did you really make two separate interfaces for the hypo and autoinjector just so you could add 4 lines to the hypospray one, with them being 95% copypaste?

Just have a variable/flag that determines if the hypo can be emptied and have that button appear in the UI conditionally.

Reply:
I uh, don't know how to do that.
I simply can't wrap my mind around conditionals in react